File: 프로그래머스/1/135808. 과일 장수/과일 장수.java
import java.util.*;

class Solution {
    public int solution(int k, int m, int[] score) {
        score = Arrays.stream(score)
            .boxed()
            .sorted(Comparator.reverseOrder())
            .mapToInt(Integer::intValue)
            .toArray();
        
        int pointer = 0;
        int answer = 0;
        
        while(pointer + m <= score.length) {
            for (int i = pointer; i < pointer + m; i++) {
                if (score[i] < k) {
                    k = score[i];
                }
            }
            
            answer += k * m;
            pointer += m;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/135808. 과일 장수/과일 장수.kt
class Solution {
    fun solution(k: Int, m: Int, score: IntArray): Int {
        score.sort()
        score.reverse()

        var answer = 0
        
        for (i in m - 1 until score.size step m) {
            answer += score[i] * m
        }
        
        return answer
    }
}